app/scaffold-controllers/navigationHeaderController.js

```diff
@@ -21,20 +25,36 @@ function(
     };
 
     NavigationHeaderController.init = function(counterBadgeController) {
-        return HeaderBarPlugin.init().then(function(headerBar) {
+        var generateSearchIcon = function() {
+            return ImageViewPlugin.init().then(function(searchIcon) {
+                searchIcon.setImagePath(HeaderConfig.searchHeaderContent.imageUrl);
+                return searchIcon;
+            });
+        };
+
+        var generateCartIcon =
```

Comment:
Maybe we should rename this variable too seeing as now this is generating the counter badge controller.